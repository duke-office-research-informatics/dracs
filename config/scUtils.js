import React from "react";
import { ThemeProvider } from "styled-components";
import { mount, shallow } from "enzyme";
import { theme as defaultTheme } from "../lib/dracs.es.js";

export const shallowWithTheme = (children, customTheme) => {
  const theme = customTheme || defaultTheme;
  return shallow(children, { context: theme });
};

/* eslint-disable */
export function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
    ),
  });
}

//FOR ACCESSING PORTALS VIA ENZYME -- import portal from dracs
// https://github.com/airbnb/enzyme/issues/1202#issuecomment-351639783
// function mountPortal(el){
//   const wrapper =  mountWithTheme(el);
//   wrapper.setState({ visible: true});
//   const portalInstance = wrapper.find(Portal).instance();
//   return mountWithTheme(portalInstance.props.children);
// }

//FOR ACCESSING EVENT LISTENERS ATTACHED TO DOCUMENT OR DOCUMENT.BODY
//https://github.com/airbnb/enzyme/issues/426#issuecomment-296304015
// const map = {};
// document.body.addEventListener = jest.fn((event, cb) => {
//   map[event] = cb;
// });
