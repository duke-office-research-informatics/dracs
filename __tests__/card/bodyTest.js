import React from "react";
import { mountWithTheme } from "../../config/scUtils.js";
import { axe } from "jest-axe";
import { CardBody, IconMenu } from "../../lib/dracs.es.js";

describe("CardBody", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mountWithTheme(<CardBody />);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("returns no a11y violations", async () => {
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });

  it("passes className via props", () => {
    wrapper.setProps({ className: "testClass" });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("body__CardBodyWrap").hasClass("testClass")).toBe(true);
  });

  it("passes height via props", () => {
    wrapper.setProps({ height: "100px" });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("body__CardBodyWrap")).toHaveStyleRule(
      "height",
      "100px"
    );
  });

  it("renders children", () => {
    const body = mountWithTheme(
      <CardBody>
        <IconMenu />
      </CardBody>
    );
    expect(body).toMatchSnapshot();
    expect(body.find(IconMenu)).toBeDefined();
  });

  it("sets padding via props", () => {
    wrapper.setProps({ padding: "8px" });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("body__CardBodyWrap")).toHaveStyleRule(
      "padding",
      "8px"
    );
  });

  it("applies a style object via props", () => {
    wrapper.setProps({ style: { margin: "8px" } });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("body__CardBodyWrap").props().style).toHaveProperty(
      "margin",
      "8px"
    );
  });
});
