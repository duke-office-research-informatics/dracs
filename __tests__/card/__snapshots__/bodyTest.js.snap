// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBody applies a style object via props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  overflow-x: hidden;
  overflow-y: auto;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
}

.c0::-webkit-scrollbar {
  display: none;
}

<article
  className="c0"
  style={
    Object {
      "margin": "8px",
    }
  }
/>
`;

exports[`CardBody matches snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  overflow-x: hidden;
  overflow-y: auto;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
}

.c0::-webkit-scrollbar {
  display: none;
}

<article
  className="c0"
/>
`;

exports[`CardBody passes className via props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  overflow-x: hidden;
  overflow-y: auto;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
}

.c0::-webkit-scrollbar {
  display: none;
}

<article
  className="c0 testClass"
/>
`;

exports[`CardBody passes height via props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  overflow-x: hidden;
  overflow-y: auto;
  height: 100px;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
}

.c0::-webkit-scrollbar {
  display: none;
}

<article
  className="c0"
  height="100px"
/>
`;

exports[`CardBody renders children 1`] = `
.c1 {
  fill: #fff;
  vertical-align: top;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  overflow-x: hidden;
  overflow-y: auto;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
}

.c0::-webkit-scrollbar {
  display: none;
}

<article
  className="c0"
>
  <svg
    aria-label="open customize and profile menu"
    className="c1"
    color="#fff"
    height={24}
    role="img"
    viewBox="0 0 24 24"
    width={24}
  >
    <title>
      menu icon
    </title>
    <desc>
      hamburger menu icon
    </desc>
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
    />
  </svg>
</article>
`;

exports[`CardBody sets padding via props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 8px;
  overflow-x: hidden;
  overflow-y: auto;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
}

.c0::-webkit-scrollbar {
  display: none;
}

<article
  className="c0"
/>
`;
